cancermuts/datasources.py

```diff
@@ -337,6 +338,9 @@ def add_mutations(self, sequence, metadata=[]):
         if 'entrez' not in sequence.aliases.keys() or sequence.aliases['entrez'] is None:
             self.log.error('Entrez ID alias not available in sequence object')
             raise TypeError('Entrez ID alias not available in sequence object')
+        
+        if not sequence.is_canonical:
+            raise UnexpectedIsoformError("cBioPortal mutation annotation only supports canonical isoforms. Please use a canonical UniProt sequence")
```

Comment:
I think this should be the very first thing we check for, since if this is the case nothing can be done